AssertionError: isa(linfo.code,Array{Any,1})

When calling unoptimized `code_typed` on a generated function this happens:
```Julia
@generated function test()
    quote
        10 + 11
    end
end
code_typed(test, (), optimize = false)
code_typed(test, (), optimize = false)

ERROR: AssertionError: isa(linfo.code,Array{Any,1})
 in Core.Inference.InferenceState(::LambdaInfo, ::Bool, ::Bool, ::Bool) at ./inference.jl:81
 in typeinf_edge(::Method, ::Any, ::SimpleVector, ::Bool, ::Bool, ::Bool, ::Void) at ./inference.jl:1572
 in (::Core.Inference.#kw##typeinf_uncached)(::Array{Any,1}, ::Core.Inference.#typeinf_uncached, ::Method, ::Any, ::Any) at ./<missing>:0
 in #code_typed#18(::Bool, ::Function, ::Any, ::Any) at ./reflection.jl:410
 in (::Base.#kw##code_typed)(::Array{Any,1}, ::Base.#code_typed, ::Any, ::Any) at ./<missing>:0
```

This seems fixed on 0.6, but because of https://github.com/JuliaLang/julia/issues/20103 I can't get things to work there either ;)

Would be nice if we can find a fix for either one (preferably fix 0.6 :P )

--- a/test/inference.jl
+++ b/test/inference.jl
@@ -800,3 +800,12 @@ let T = Tuple{Tuple{Int64, Void},
     @test Core.Inference.limit_type_depth(T, 1) >: T
     @test Core.Inference.limit_type_depth(T, 2) >: T
 end
+
+# Issue #20902, check that this doesn't error.
+@generated function test_20902()
+    quote
+        10 + 11
+    end
+end
+@test length(code_typed(test_20902, (), optimize = false)) == 1
+@test length(code_typed(test_20902, (), optimize = false)) == 1

